vsutil.py

```diff
@@ -176,3 +195,74 @@ def is_image(filename: str, /) -> bool:
+          bitdepth: int,
+          /,
+          sample_type: Optional[Union[int, vs.SampleType]] = None,
+          *,
+          range: Optional[Union[int, Range]] = None,
+          range_in: Optional[Union[int, Range]] = None,
+          dither_type: Optional[Union[Dither, str]] = None) \
+        -> vs.VideoNode:
+    """
+    A bit depth converter only using core.resize and Format.replace.
+    By default, outputs FLOAT sample type for 32 bit and INTEGER for anything else.
+
+    :param bitdepth:    Desired bits_per_sample of output clip.
+    :param sample_type: Desired sample_type of output clip. Allows overriding default FLOAT/INTEGER behavior. Accepts
+                        vapoursynth.SampleType enums INTEGER and FLOAT or their values, [0, 1].
+    :param range:       Output pixel range (defaults to input clip's range). See `Range`.
+    :param range_in:    Input pixel range (defaults to input clip's range). See `Range`.
+    :param dither_type: Dithering algorithm. Allows overriding default dithering behavior. See `Dither`.
+        Defaults to Floyd-Steinberg error diffusion when downsampling, converting between ranges, or upsampling full
+        range input. Defaults to 'none', or round to nearest, otherwise.
+
+    :return: Converted clip with desired bit depth and sample type. ColorFamily will be same as input.
+    """
+    sample_type = _resolve_enum(vs.SampleType, sample_type, 'sample_type', vs.__name__)
+    range = _resolve_enum(Range, range, 'range')
+    range_in = _resolve_enum(Range, range_in, 'range_in')
+    dither_type = _resolve_enum(Dither, dither_type, 'dither_type')
+
+    curr_depth = get_depth(clip)
+    sample_type = fallback(sample_type, vs.FLOAT if bitdepth == 32 else vs.INTEGER)
+
+    if (curr_depth, clip.format.sample_type, range_in) == (bitdepth, sample_type, range):
+        return clip
+
+    # thanks @Frechdachs for explaining this:
+    # 'you need dithering when raising the bitdepth of full range [or] converting between full and limited'
+    should_dither = (range_in != range
+                     or range_in == Range.FULL
+                     or (curr_depth > bitdepth and sample_type == vs.INTEGER))
+    dither_type = fallback(dither_type, Dither.ERROR_DIFFUSION if should_dither else Dither.NONE)
+
+    return clip.resize.Point(format=clip.format.replace(bits_per_sample=bitdepth, sample_type=sample_type),
+                             range=range,
+                             range_in=range_in,
+                             dither_type=dither_type.value)
+
+
+def _readable_enums(enum: EnumMeta, module: str = 'vsutil') -> list:
+    """
+    Returns a list of all possible values in `module.enum`.
+    Extends the default `repr(enum.value)` behavior by prefixing the enum with the name of the module it belongs to.
+    """
+    return [f'<{module}.{str(e)}: {e.value}>' for e in enum]
+
+
+def _resolve_enum(enum: EnumMeta, value: Any, var_name: str, module: str = 'vsutil'):
```

Comment:
See comment on the signature of `_readable_enums`.

Reply:
```suggestion
def _resolve_enum(enum: EnumMeta, value: Any, var_name: str, module: str = 'vsutil') -> Any:
```
I think this is also relevant but not sure if there's a better way of handling this.

Reply:
So I found a proper solution to the return type:
```py
E = TypeVar('E', bound=EnumMeta)
```
```suggestion
def _resolve_enum(enum: EnumMeta, value: Any, var_name: str, module: str = 'vsutil') -> Union[E, None]:
```

Reply:
Discussed with Fichte on TeamSpeak, https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/29/commits/616457b1472029aa8db63939bd0eb1b9c9e8de09 resolves these issues.